### PR TITLE
Move to TypeScript

### PR DESCRIPTION
**Pull request by avalynndev**
a release that updates the code from JavaScript to typescript.

# Requirements

- [x] The base branch is pointing to the correct branch. 

## Codebase

- [x] This code has been tested with the [latest release](https://github.com/EasyScriptJS/EasyScript/releases/latest) of Easy Script.
- [x] This code works with the [latest release](https://github.com/EasyScriptJS/EasyScript/releases/latest) of Easy Script.
- [x] This code has been run through [Prettier](https://prettier.io).

### Modules

- [x] I have provided documentation of how this code works. 

# Description

Converts the code from JavaScript to typescript. Adds new features, making it more useful

### CODE REVIEW

**Review by wdhdev — CHANGES_REQUESTED** [review]
Can you please run the code through prettier replacing semi colons with regular colons.

aka ' -> "

**Comment by avalynndev** [comment]
I don't understand what you mean. Please send an example.

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/** Converting Modules **/
+
+export { boolToNum } from "./src/types/boolToNum";
+export { boolToStr } from "./src/types/boolToStr";
+export { numToBool } from "./src/types/numToBool";
+export { numToStr } from "./src/types/numToStr";
+export { strToBool } from "./src/types/strToBool";
+export { strToNum } from "./src/types/strToNum";
+
+/** Others **/
```
Comment (wdhdev) [inline index.ts]:
```suggestion
/** Util **/
```

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/** Modules **/
+
+export { flip } from "./src/flip";
+export { log } from "./src/log";
+export { print } from "./src/print";
+export { random } from "./src/random";
+export { uuid } from "./src/uuid";
+
+/** Converting Modules **/
```
Comment (wdhdev) [inline index.ts]: Please move this section into util section as these are no longer located in the types module

File: src/flip.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (wdhdev) [inline src/flip.ts:1]: This file has been updated since

File: src/log.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
```
Comment (wdhdev) [inline src/log.ts:1]: This file has been updated since

File: src/print.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @function `print` - Quickly log any data to the console.
```
Comment (wdhdev) [inline src/print.ts:2]: This file has been updated since

File: src/random.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/**
```
Comment (wdhdev) [inline src/random.ts:1]: This file has been updated since

File: src/types/boolToNum.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export function boolToNum(boolean: boolean): number {
```
Comment (wdhdev) [inline src/types/boolToNum.ts:1]: This file has been updated since to include self docs.

File: src/types/numToBool.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export function numToBool(number: number): boolean {
```
Comment (wdhdev) [inline src/types/numToBool.ts:1]: This file has been updated since to include self docs.

File: src/types/numToStr.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export function numToStr(number: number): string {
```
Comment (wdhdev) [inline src/types/numToStr.ts:1]: This file has been updated since to include self docs.

File: src/types/strToBool.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export function strToBool(string: string): boolean {
```
Comment (wdhdev) [inline src/types/strToBool.ts:1]: This file has been updated since to include self docs.

File: src/types/strToNum.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+export function strToNum(string: string): number {
```
Comment (wdhdev) [inline src/types/strToNum.ts:1]: This file has been updated since to include self docs.

File: src/util/removeDuplicates.ts [hunk 12]
```diff
@@ -0,0 +1,135 @@
+import { ObjectType } from "typescript";
```
Comment (wdhdev) [inline src/util/removeDuplicates.ts:1]: This file has been updated since

File: src/uuid.ts [hunk 13]
```diff
@@ -0,0 +1,21 @@
+/**
```
Comment (wdhdev) [inline src/uuid.ts:1]: This file has been updated since

**Review by wdhdev — CHANGES_REQUESTED** [review]
Please also move modules back into `/modules` folder

**Pull request merged**